packages/boot/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {statSync, readdirSync} from 'fs';
+import {join, resolve} from 'path';
+import {StringOrArr} from './types';
+const debug = require('debug')('@loopback/boot:utils');
+
+/**
+ * Takes an absolute directory path and returns all files / folders in it. Can optionally
+ * traverse nested folders.
+ *
+ * @param d Absolute path of directory to read
+ * @param nested Read nested directories in the given directory or not. Defaults to true.
+ */
+// tslint:disable-next-line:no-any
+export function readFolder(d: string, nested: boolean = true): any {
+  // If not reading nested directories, read the current one and return
+  if (!nested) {
+    try {
+      let files = readdirSync(d).map(f => join(d, f));
+      return files;
+    } catch (err) {
+      debug(`Skipping ${d} in if(!recursive) because error: ${err}`);
```

Comment:
We need to take more care of error handling and carefully consider which errors are safe to hide to a debug log and which must be propagated back to the caller.

For example, when there is no `controllers` directory, then it makes sense to me to return an empty array and leave a message in the debug log.

However, if the `controllers` directory is there, but we were not able to read its contents, then that's a serious error that should abort the boot process.